src/Libraries/RevitNodes/Elements/Roof.cs

```diff
@@ -0,0 +1,289 @@
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Create a Revit Roof given it's curve outline and Level
+        /// </summary>
+        /// <param name="outline"></param>
+        /// <param name="RoofType"></param>
+        /// <param name="level"></param>
+        /// <returns>The Roof</returns>
+        public static Roof ByOutlineTypeAndLevel(Curve[] outline, RoofType roofType, Level level)
+        {
+            var polycurve = PolyCurve.ByJoinedCurves(outline);
+
+            if (!polycurve.IsClosed)
+            {
+                throw new ArgumentException(Properties.Resources.OpenInputPolyCurveError);
+            }
+
+            var ca = new CurveArray();
+            polycurve.Curves().ForEach(x => ca.Append(x.ToRevitType()));
+
+            return new Roof(ca, level.InternalLevel,roofType.InternalRoofType);
+        }
+
+        /// <summary>
+        /// Extrude Roof by Outline, Referenceplane
+        /// </summary>
+        /// <param name="outline"></param>
+        /// <param name="roofType"></param>
+        /// <param name="level"></param>
+        /// <param name="plane"></param>
+        /// <param name="extrusionStart"></param>
+        /// <param name="extrusionEnd"></param>
+        /// <returns></returns>
+        public static Roof ByOutlineExtrusionTypeAndLevel(PolyCurve outline, RoofType roofType, Level level, ReferencePlane plane, double extrusionStart, double extrusionEnd)
+        {
+
+            if (!outline.IsClosed)
+            {
+                throw new ArgumentException(Properties.Resources.OpenInputPolyCurveError);
+            }
+
+            var ca = new CurveArray();
+            outline.Curves().ForEach(x => ca.Append(x.ToRevitType()));
+
+            return new Roof(ca, plane.InternalReferencePlane, level.InternalLevel, roofType.InternalRoofType, extrusionStart, extrusionEnd);
+        }
+
+
+        #endregion
+
+        #region Public Properties
+
+        /// <summary>
+        /// Get Slab Shape Points
+        /// </summary>
+        public IEnumerable<Pt> Points
+        {
+            get 
+            {
+                if (this.InternalRoof.SlabShapeEditor == null)
+                {
+                    throw new Exception(Properties.Resources.InvalidShapeEditor);
+                }
+
+                TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+                this.InternalRoof.SlabShapeEditor.Enable();
+                TransactionManager.Instance.TransactionTaskDone();
+
+                List<Pt> points = new List<Pt>();
+                foreach (SlabShapeVertex v in this.InternalRoof.SlabShapeEditor.SlabShapeVertices)
+                {
+                    points.Add(v.Position.ToPoint());
+                }
+                return points;
+            }
+        }
+
+        /// <summary>
+        /// Add Point to Slab Shape
+        /// </summary>
+        public void AddPoint(Pt point)
```

Comment:
Similarly it is better to return a roof for this method and the MovePoint method.